coreapi.go

```diff
@@ -22,6 +23,8 @@ type CoreAPI interface {
 	// Dag returns an implementation of Dag API
 	Dag() APIDagService
 
+	Node() NodeAPI
```

Comment:
There aren't any interface tests for the new interfaces added in this PR. How much work do you think it is to add relevant tests?

Reply:
@aschmahmann if you are ok removing the new API, this would be my preference for the first merge. I would rather hold off on a public endpoint until we have more usage of these two modules internal to the stack (Fetcher and Dag writer). While they're tagged public modules with public methods in their own repos, that's less of a commitment than exposing them on this API -- cause when we do that, we're basically locking ourselves in. What do you think?

Reply:
If you're ok, I'll revert that part of this PR.

Reply:
yep, let's just not do this now. We don't have a story for how we want the HTTP API to handle this anyway so let's just file an issue and tackle this later.

It's a bit of a shame that it means we won't be able to tell consumers they can totally drop go-ipld-format, but that'll just have to come later.

Reply:
> I'll revert that part of this PR.

IIRC that makes this PR basically dependency updates + a test fix rather than anything about IPLD + IPFS in particular. If so then maybe just make that it's own unrelated PR. If this is needed for the other work to pass or something then just leave it in this PR.